docs/hardware/3-esp32/1-espidfquickstart/3-flash-sample.md

```diff
@@ -0,0 +1,92 @@
+---
+id: flash-sample
+title: Flashing with samples
+---
+
+Now we're getting to the good stuff - working with hardware! The sample project
+provided will demonstrate a device securely communicating with Golioth and will
+demonstrate core features like LightDB State/Stream, Logging, and OTA Updates.
+
+### Build the `golioth_basics` code example
+
+Golioth ESP-IDF SDK example code is located in the
+`golioth-esp-idf-sdk/examples`. Navigate to the `golioth_basics` example and run
+the following code to build the example app:
+
+```console
+idf.py set-target esp32
+idf.py build
+```
+
+### Flashing the device
+
+Flashing is a simple `idf.py` command away.
+
+```console
+idf.py flash
+```
+
+:::note
+If your ESP32 is not found by the flash command, you may need to specify
+a port such as: `idf.py -p /dev/ttyUSB0 flash`
+:::
+
+### Set credentials in the serial terminal
+
+You must set WiFi and Golioth credentials for the example to connect properly.
+Connect to the device over a serial console using a tool like `screen`:
+
+```console
+screen /dev/ttyUSB0 115200
```

Comment:
It might be a good idea to recommend `idf.py monitor` instead. It's easier, and I think it will do extra stuff for you, like if the board crashes, it will decode the raw backtrace data and print the human-readable backtrace (you won't get this with just a serial terminal).